File: blackjack/csharp/Card.cs
public class Card
{
    private char suit;
    private string value;

    // Construct a card with a given suit and value
    public Card(string value, char suit) {
        this.suit = suit;
        this.value = value;
    }


    // Retrieve the value
    public string getValue() {
        return value;
    }


    // Retrieve the suit
    public char getSuit() {
        return suit;
    }


    // Convert the card to a string for easy printing
    public override string ToString() {
        return String.Format("{0,2}{1}", getValue(), getSuit());
    }
}

File: blackjack/ruby/card.rb
class Card
  # A card is initialized with a given suit and value
  def initialize(value, suit)
    # store the class variables, convert to string
    @value = value.to_s
    @suit = suit.to_s
  end


  # Convert the card to a string suitable for printing
  def to_s
    return "%2s%s"%[value, suit]
  end


  # read only accessors
  attr_reader :value
  attr_reader :suit
end
